### PR TITLE
SSV contract changes

### PR DESCRIPTION
**Pull request by naddison36**
# Contract Changes

* Changed `stakeETH` on `NodeDelegator` so it takes an array of validator data. This is used to stake WETH or ETH in the NodeDelegator to multiple validators connected to the associated EigenPod.
* Added the following functions to the `NodeDelegator` for SSV support
  - `approveSSV()`
  - `depositSSV(uint64[] memory operatorIds, uint256 amount, Cluster memory cluster)`
  - `registerSsvValidator(bytes calldata publicKey,
        uint64[] calldata operatorIds,
        bytes calldata sharesData,
        uint256 amount,
        Cluster calldata cluster)`
* Access controls are:
  * Manager: approveSSV and depositSSV
  * Operator: registerSsvValidator

# SSV Processes
primeSSVRestakingProcess

### CODE REVIEW

File: contracts/NodeDelegator.sol [hunk 1]
```diff
@@ -29,6 +29,12 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
 
     uint256 internal constant DUST_AMOUNT = 10;
 
+    struct Validator {
```
Comment (sparrowDom) [inline contracts/NodeDelegator.sol:32]: nit: maybe ValidatorDepositData would be a better name for this to be more aligned with P2P terminology. The signature and depositDataRoot only pertain to depositing ETH. The pubkey is a permanent property of the validator

**Review by sparrowDom — APPROVED** [review]
looks good

File: contracts/NodeDelegator.sol (hunk 1 above)
Comment (naddison36) [reply contracts/NodeDelegator.sol:33]: I've renamed to `ValidatorStakeData`

File: contracts/NodeDelegator.sol [hunk 2]
```diff
@@ -243,13 +256,14 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
...
+    /// @notice Stake WETH or ETH in NDC to a validator connected to an EigenPod.
+    /// This function calls `stake` on the EigenPodManager which calls `stake` on the EigenPod contract which calls
+    /// `stake` on the Beacon DepositContract.
     /// @param pubkey The pubkey of the validator
     /// @param signature The signature of the validator
     /// @param depositDataRoot The deposit data root of the validator
-    /// @dev Only LRT Operator should call this function
-    /// @dev Exactly 32 ether is allowed, hence it is hardcoded
+    /// @dev Only accounts with the Operator role can call this function.
+    /// @dev Exactly 32 ether is allowed, hence it is hardcoded.
```
[the remark this answers is not in the document]
Comment (naddison36) [reply contracts/NodeDelegator.sol]: I've consolidated the `stakeETH` and `bulkStakeETH` functions to just `stakeETH` that accepts an array of validators.

**Pull request merged**